carma-messenger-core/cpp_message/src/MobilityRequest_Message.cpp

```diff
@@ -474,17 +434,16 @@ namespace cpp_message
             return boost::optional<std::vector<uint8_t>>{};
         }
 
-        MobilityLocationOffsets* offsets_list;
-        offsets_list=(MobilityLocationOffsets*)calloc(1,sizeof(MobilityLocationOffsets));
-        if(!offsets_list)
+        std::shared_ptr <MobilityLocationOffsets>offsets_list_shared ((MobilityLocationOffsets*)calloc(1,sizeof(MobilityLocationOffsets)),free);
+        if(!offsets_list_shared)
         {
             ROS_WARN_STREAM("Cannot allocate mem for offsets list encoding");
             return boost::optional<std::vector<uint8_t>>{};
         }
+        MobilityLocationOffsets* offsets_list=offsets_list_shared.get();    
 
-        MobilityECEFOffset* Offsets;    
         for(size_t i=0;i<offset_count;i++){
-            Offsets=(MobilityECEFOffset*)calloc(1,sizeof(MobilityECEFOffset));
+            MobilityECEFOffset* Offsets=new MobilityECEFOffset;
```

Comment:
Hmm, does this underlying memory get freed at some point? I don't recall off the top of my head if `delete`ing a vector of pointers (as I'm assuming shared_ptr does when it destroys the message it points to) actually frees the items the pointers point at. Might be worth a quick look to verify. If you're familiar with a tool like `valgrind` you could use that on your unit test executable to see if it detects any leaks. If not, I wouldn't worry about it for now and just see if you can figure out what's supposed to happen to try to make sure these allocations don't leak.